File: apzkr-pzpi-21-11-milennyi-ivan/Task3-WebClient/src/models/Sheep.ts
// Клас, що представляє вівцю
export class Sheep {
    id: number | null; // Ідентифікатор вівці
    birth_date: number; // Дата народження вівці у секундах
    breed_id: number; // Ідентифікатор породи
    weight: number; // Вага вівці
    sex: boolean; // Стать вівці (true - самець, false - самка)
    temperature_scanner_id: number | null; // Ідентифікатор температурного сканера
    shepherd_id: number | null; // Ідентифікатор пастуха

    constructor(id: number | null, birthDate: number, breedId: number, weight: number, sex: boolean, temperatureScannerId: number | null, shepherdId: number | null) {
        this.id = id;
        this.birth_date = birthDate;
        this.breed_id = breedId;
        this.weight = weight;
        this.sex = sex;
        this.temperature_scanner_id = temperatureScannerId;
        this.shepherd_id = shepherdId;
    }
}

File: apzkr-pzpi-21-11-milennyi-ivan/Task1-Server/src/models/sheep.rs
use serde::{Deserialize, Serialize};
use sqlx::FromRow;
use utoipa::ToSchema;
use validator::Validate;

// Структура, що представляє вівцю
#[derive(Debug, Validate, Deserialize, Serialize, FromRow, ToSchema)]
pub(crate) struct Sheep {
    id: Option<u64>, // Ідентифікатор вівці
    birth_date: u64, // Дата народження (у вигляді таймстемпу)
    breed_id: u64, // Ідентифікатор породи
    weight: u64, // Вага
    sex: bool, // Стать
    temperature_scanner_id: Option<u64>, // Ідентифікатор сканера температури
    shepherd_id: Option<u64> // Ідентифікатор пастуха
}

impl Sheep {
    // Отримання ідентифікатора вівці
    pub fn id(&self) -> Option<u64> {
        self.id
    }

    // Встановлення ідентифікатора вівці
    pub fn set_id(&mut self, id: u64) -> () {
        self.id = Some(id);
    }

    // Отримання дати народження
    pub fn birth_date(&self) -> u64 {
        self.birth_date
    }

    // Отримання ідентифікатора породи
    pub fn breed_id(&self) -> u64 {
        self.breed_id
    }

    // Отримання ваги
    pub fn weight(&self) -> u64 {
        self.weight
    }

    // Отримання статі
    pub fn sex(&self) -> bool {
        self.sex
    }

    // Отримання ідентифікатора сканера температури
    pub fn temperature_scanner_id(&self) -> Option<u64> {
        self.temperature_scanner_id
    }

    // Отримання ідентифікатора пастуха
    pub fn shepherd_id(&self) -> Option<u64> {
        self.shepherd_id
    }
}
